tpp-app/src/onboarding/Confirmation.js

```diff
@@ -20,7 +23,54 @@ export const STACK_SCREENS = {
   CONFIRMATION : "Confirmation"
 };
 
-export default function Confirmation ({ navigation }) {
+export default function Confirmation ({ route, navigation }) {
+  const { periodLength, periodStart, periodEnd, trackingPreferences } = route.params;
+
+  function FlowPref() {
+    if(trackingPreferences[0])
+      return (<FlowIcon style={styles.icon}/>);
+    else
+      return (null);
+  }
+  function MoodPref() {
+    if(trackingPreferences[1])
+      return (<MoodIcon style={styles.icon}/>);
+    else
+      return (null);
+  }
+  function SleepPref() {
+    if(trackingPreferences[2])
+      return (<SleepIcon style={styles.icon}/>);
+    else
+      return (null);
+  }
+  function CrampsPref() {
+    if(trackingPreferences[3])
+      return (<CrampsIcon style={styles.icon}/>);
+    else
+      return (null);
+  }
+  function ExercisePref() {
+    if(trackingPreferences[4])
+      return (<ExerciseIcon style={styles.icon}/>);
+    else
+      return (null);
```

Comment:
This is a bit repetitive but I also ran into the same issue with icons so I think it's fine

Reply:
Could these be put into a switch case "Pref" function that takes a parameter and defaults to null if we wanted to make it more concise

Reply:
oops i was being lazy on this and told myself i would come back but i guess i never did - fixed now!